### PR TITLE
[ISSUE #8829] feat: provide ConfigManagerV2 to make best uses of RocksDB

### PR DESCRIPTION
**Pull request by lizhanhui**
### Which Issue(s) This PR Fixes

To #8829

### CODE REVIEW

File: broker/src/main/java/org/apache/rocketmq/broker/config/v2/SubscriptionGroupManagerV2.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    @Override
+    public void updateSubscriptionGroupConfig(final SubscriptionGroupConfig config) {
+        if (config == null || MixAll.isLmq(config.getGroupName())) {
+            return;
+        }
+        ByteBuf keyBuf = ConfigHelper.keyBufOf(TableId.SUBSCRIPTION_GROUP, config.getGroupName());
+        ByteBuf valueBuf = ConfigHelper.valueBufOf(config, SerializationType.JSON);
+        try {
+            WriteBatch writeBatch = new WriteBatch();
```
Comment (redlsz) [inline broker/src/main/java/org/apache/rocketmq/broker/config/v2/SubscriptionGroupManagerV2.java:137]: Need to release the wirteBatch underlying resoueces by calling close() or using try-with-resources.

Comment (lizhanhui) [reply broker/src/main/java/org/apache/rocketmq/broker/config/v2/SubscriptionGroupManagerV2.java:137]: Good catch

**Review by RongtongJin — COMMENTED** [review]
Hi, @lizhanhui , we have been using version v1 in the production environment,  and it's likely that some community users are doing the same. From a code perspective, a compatibility upgrade from v1 to v2 is not feasible. Given the need for in-place upgrade, I think it's essential to add support for persisting lmq consumer offsets in version v1.

**Comment by lizhanhui** [comment]
> Hi, @lizhanhui , we have been using version v1 in the production environment, and it's likely that some community users are doing the same. From a code perspective, a compatibility upgrade from v1 to v2 is not feasible. Given the need for compatibility upgrade, I think it's essential to add support for persisting lmq consumer offsets in version v1.

Let's do this in the next pull request. This pull request has been kind of too large.

**Comment by tianliuliu** [comment]
LGTM

**Review by tianliuliu — APPROVED** [review]

**Review by fuyou001 — APPROVED** [review]

**Pull request merged**